apps/aechannel/src/aesc_fsm.erl

```diff
@@ -1535,7 +1535,6 @@ new_onchain_tx(channel_withdraw_tx, #{acct := ToId,
                #data{on_chain_id = ChanId, state=State} = D,
                BlockHash, OnChainEnv, OnChainTrees) ->
     Updates = [aesc_offchain_update:op_withdraw(aeser_id:create(account, ToId), Amount)],
-    {BlockHash, OnChainEnv, OnChainTrees} = pick_onchain_env(Opts, D),
```

Comment:
This function already receives the env as arguments. Fetching it *again* is wrong and potentially it could yeld different results. Especially in a fast mining system-test setup, that is.